libs/shared/providers/src/notifications/components/NotificationSnack.tsx

```diff
@@ -0,0 +1,83 @@
+import {
+  Alert,
+  AlertTitle,
+  IconButton,
+  SvgIcon,
+  Typography,
+} from '@mui/material';
+import { isNilOrEmpty } from '@origin/shared/utils';
+import { useIntervalEffect } from '@react-hookz/web';
+
+import { useSetNotificationInvisible, useSetNotificationRead } from '../hooks';
+
+import type { AlertProps, AlertTitleProps } from '@mui/material';
+import type { Dispatch } from 'react';
+
+import type { Notification } from '../types';
+
+export type NotificationSnackProps = {
+  notification: Notification;
+  onClose?: Dispatch<Notification>;
+  AlertProps?: Omit<AlertProps, 'severity'>;
+  AlertTitleProps?: Omit<AlertTitleProps, 'children' | 'title'>;
+};
+
+export const NotificationSnack = ({
+  notification,
+  onClose,
+  AlertProps,
+  AlertTitleProps,
+}: NotificationSnackProps) => {
+  const { id, severity, title, message, content } = notification;
+  const setNotificationViewed = useSetNotificationRead();
+  const setNotificationInvisible = useSetNotificationInvisible();
+
+  useIntervalEffect(() => {
+    setNotificationInvisible(id);
+  }, notification.hideDuration);
+
+  const handleCloseClick = () => {
+    if (onClose) {
+      onClose(notification);
+    }
+    setNotificationViewed(id);
+    setNotificationInvisible(id);
+  };
+
+  return (
+    <Alert
+      {...AlertProps}
+      severity={severity}
+      sx={{
+        border: 1,
+        borderRadius: 1,
+        minWidth: { sm: 300, md: 400, lg: 500, xl: 600 },
+        maxWidth: { sm: 400, md: 500, lg: 600, xl: 700 },
+        ...AlertProps?.sx,
+      }}
+      action={
+        <IconButton size="small" aria-label="close" onClick={handleCloseClick}>
+          <SvgIcon>
```

Comment:
do you need to restyle the svg programatically? I would rather put the svg in `shared/asset` and load it as an image. this way the svg is not part of the js file and it can be cached in the browser as an image

Reply:
yeah, I saw that you were doing that, I think we still want to settle for an icon library and it would remove the need for this, but, I'm fine going for yous solution in the meantime